File: src/time_utils.py
import time
import datetime as dt

def get_time():
    #t = dt.datetime.now(dt.UTC)
    #t = t.timestamp() * 1_000_000
    #return int(round(t))
    return time.time_ns()

def get_datetime():
    return dt.datetime.now(dt.UTC)

File: java_code/write-update/src/main/java/project/TimeUtils.java
package project;

import java.time.Clock;
import java.time.Instant;
import java.time.LocalDateTime;
import java.time.ZoneOffset;

public class TimeUtils {
	
    public static long getTime() {
    	Instant now = Instant.now(Clock.systemUTC());
        long seconds = now.getEpochSecond();
        long nanos = now.getNano();
        return seconds * 1_000_000_000L + nanos;
    }

    public static LocalDateTime getDatetime() {
        return LocalDateTime.now(ZoneOffset.UTC);
    }

    public static void main(String[] args) {
        System.out.println("Current time in nanoseconds: " + getTime());
        System.out.println("Current datetime in UTC: " + getDatetime());
    }
}
